packages/rum-core/src/domain/rumEventsCollection/view/trackInteractionToNextPaint.ts

```diff
@@ -0,0 +1,113 @@
+import { type Duration, noop, isExperimentalFeatureEnabled, ExperimentalFeature } from '@datadog/browser-core'
+import { supportPerformanceTimingEvent, type RumEventTiming } from '../../../browser/performanceCollection'
+import { LifeCycleEventType, type LifeCycle } from '../../lifeCycle'
+import { ViewLoadingType } from '../../../rawRumEvent.types'
+import { getInteractionCount, initInteractionCountPolyfill } from './interactionCountPolyfill'
+
+// Arbitrary value to prevent unnecessary memory usage on views with lots of interactions.
+const MAX_INTERACTION_ENTRIES = 10
+
+/**
+ * Track the interaction to next paint (INP).
+ * To avoid outliers, return the p98 worst interaction of the view.
+ * Documentation: https://web.dev/inp/
+ * Reference implementation: https://github.com/GoogleChrome/web-vitals/blob/main/src/onINP.ts
+ */
+export function trackInteractionToNextPaint(viewLoadingType: ViewLoadingType, lifeCycle: LifeCycle) {
+  if (
+    !isInteractionToNextPaintSupported() ||
+    !isExperimentalFeatureEnabled(ExperimentalFeature.INTERACTION_TO_NEXT_PAINT)
+  ) {
+    return {
+      getInteractionToNextPaint: () => undefined,
+      stop: noop,
+    }
+  }
+
+  const { getViewInteractionCount } = trackViewInteractionCount(viewLoadingType)
+  const longestInteractions = trackLongestInteractions(getViewInteractionCount)
+  let maxInpDuration = -1 as Duration
+
+  const { unsubscribe: stop } = lifeCycle.subscribe(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, (entries) => {
+    for (const entry of entries) {
+      if (entry.entryType === 'event' && entry.interactionId) {
+        longestInteractions.process(entry)
+      }
+    }
+
+    const inp = longestInteractions.estimateP98Duration()
+    if (inp && inp.duration > maxInpDuration) {
+      maxInpDuration = inp.duration
+    }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm not questioning your implementation, but using a max of a p98 feels weird to me. I believe using a percentile here helps to get rid of outliers . Imagine the first interaction you have is an outlier and its value is abnormally high, `maxInpDuration` will be set to this value. Then, we have a thousand more interaction with a smaller duration. `maxInpDuration` will still represent the first outlier, defeating the purpose of using a percentile.

Reply:
Ah I wasn't really suggesting to change the implementation, sorry :D I'm not sure what's the right call here, we might want to stick with Google approach.

Reply:
No it's not the Google approach it's a wrong implementation on my side while trying to make INP for every view